### PR TITLE
FIX: several fixes for pyart.aux_io.read_gamic()

### PR DESCRIPTION
**Pull request by kmuehlbauer**
This fixes the `pulse_width_us` vs `pulse_width_mks` described in #534

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
keying into wrong group, fixed, sorry

**Comment by kmuehlbauer** [comment]
- added the missing moments to the config file
- added fix for `az_speed` vs. `azimuth_speed` naming issue
- check for nyquist_velocity attribute in `extended`
- added hack for getting `pulse_width` enum correctly for BoXPol gamic files

The latter should be made versatile via config-file, but I've no clue how to to this. 

For testing purposes you can find here [PPI](https://github.com/wradlib/wradlib-data/blob/master/hdf5/2014-08-10--182000.ppi.mvol) and [RHI](https://github.com/wradlib/wradlib-data/blob/master/hdf5/2014-06-09--185000.rhi.mvol) scans.

File: pyart/aux_io/gamic_hdf5.py [hunk 1]
```diff
@@ -277,8 +287,17 @@ def _get_instrument_params(gfile, filemetadata):
...
-    dic['data'] = gfile.sweep_expand(
-        gfile.how_attrs('pulse_width_us', 'float32') * 1e-6)
+    pw_names = ['pulse_width_us', 'pulse_width_mks', 'pulse_width']
+    pw_name = 'pulse_width_us'
+    for pw_name in pw_names:
+        if gfile.is_attr_in_group('/scan0/how', pw_name):
+            break
+    if pw_name == 'pulse_width':
+        dic['data'] = gfile.sweep_expand(
+        [0.2, 0.5, 1.0, 2.0][gfile.how_attrs(pw_name, 'int')] * 1e-6)
```
Comment (kmuehlbauer) [inline pyart/aux_io/gamic_hdf5.py]: These are the pulsewidth's (in us) for the BoxPol Radar.

File: pyart/aux_io/gamic_hdf5.py [hunk 2]
```diff
@@ -300,8 +319,9 @@ def _get_instrument_params(gfile, filemetadata):
     instrument_params['unambiguous_range'] = dic
 
     dic = filemetadata('nyquist_velocity')
```
Comment (jjhelmus) [inline pyart/aux_io/gamic_hdf5.py]: I would move this inside the if block, there is no reason to create the dictionary if it is not used.

File: pyart/aux_io/gamic_hdf5.py [hunk 3]
```diff
@@ -300,8 +319,9 @@ def _get_instrument_params(gfile, filemetadata):
...
-    dic['data'] = gfile.sweep_expand(gfile.how_ext_attrs('nyquist_velocity'))
-    instrument_params['nyquist_velocity'] = dic
+    if gfile.is_attr_in_group('/scan0/how/extended', 'nyquist_velocity'):
+        dic['data'] = gfile.sweep_expand(gfile.how_ext_attrs('nyquist_velocity'))
```
Comment (jjhelmus) [inline pyart/aux_io/gamic_hdf5.py]: This line is a bit toi long, splitting in into two lines would be good and could improve readability.

File: pyart/aux_io/gamic_hdf5.py (hunk 1 above)
Comment (jjhelmus) [inline pyart/aux_io/gamic_hdf5.py]: With the latest version of NumPy this line is raising DeprecationWarning:

```
/home/jhelmus/dev/pyart/pyart/aux_io/gamic_hdf5.py:297: DeprecationWarning: converting an array with ndim > 0 to an index will result in an error in the future
  [0.2, 0.5, 1.0, 2.0][gfile.how_attrs(pw_name, 'int')] * 1e-6)
```
Selecting the first element of the extracted slice solves this:
```
[0.2, 0.5, 1.0, 2.0][gfile.how_attrs(pw_name, 'int')[0]] * 1e-6)
```

**Comment by jjhelmus** [comment]
This looks good @kmuehlbauer, thanks for putting this together.  I made a few recommendation for improvement, then this should be good to merge.

I did notice that the gamic modules are broken in Python 3, although not a result of any changes you made.  I will work on getting these to work in Python 3 once this gets merged.

**Comment by kmuehlbauer** [comment]
@jjhelmus I'll incorporate everything by tomorrow. Do you have any idea, how we could make the pulsewidth list a parameter or a config-file member. Normally I'm against hardcoding stuff. This might work then only for the two radars here, but there might be others too, which have different pulsewidth'. 

If you have some neat idea, I'll gladly fit this in.

**Comment by jjhelmus** [comment]
> Do you have any idea, how we could make the pulsewidth list a parameter or a config-file member.

I'm not sure if this information should be encoding in the configuration file since it is radar specific.  Would adding it as an optional keyword argument to the `read_gamic` function work?  I have not work enough with GAMIC files to know how different these values tend to be be.

**Comment by kmuehlbauer** [comment]
> Would adding it as an optional keyword argument to the read_gamic function work? I have not work enough with GAMIC files to know how different these values tend to be be.

I've really no idea, what radar pulsewidth gamic radar processors are dealing with, but we should assume that pw are not like ours. So I like the idea of having this as keyword argument of the read_gamic function with an additional error message if `pulse_width` is given while keyword is missing. You know what I mean. I'll try to get something working.

**Comment by kmuehlbauer** [comment]
@jjhelmus I added an parameter to `read_gamic()` which defaults to None. Added also to the docstring. Works here fine for my use-case and should also not distrub any other gamic-flavours.

**Comment by jjhelmus** [comment]
Thanks for the updates. Looks good to me and is able to open all the gamic files I have laying around.  I'll fix up Python 3 support in a separate PR shortly.

**Pull request merged**